src/vl53l0x_driver.cpp

```diff
@@ -1,185 +1,149 @@
+#include <cstring>
 #include <ros/ros.h>
-#include "std_msgs/Int16.h"
-#include <stdlib.h>
+#include <std_msgs/Int16.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 
-extern "C" {  
-    #include "mcp23017.h"  
-}  
+extern "C" {
+#include "mcp23017.h"
+}
 
 #include "vl53l0x_api.h"
-#include "vl53l0x_platform.h"
-
-#include "vl53l0x_driver/getSensorData.h"
 #include "vl53l0x_driver/vl53l0x.h"
- 
+#include "vl53l0x_platform.h"
 #define MCP_ADDRESS 0x20
-#define I2C_BUS_PATH   "/dev/i2c-8"
-#define I2C_BUS_INSTANCE   8
-
-#define VL53L0X_1_XSHUT_MCP23xx_IO  0
-#define VL53L0X_2_XSHUT_MCP23xx_IO  1
-#define VL53L0X_3_XSHUT_MCP23xx_IO  2
-#define VL53L0X_4_XSHUT_MCP23xx_IO  3
-
-#define VL53L0X_DEFAULT_ADDR  0x29
-#define VL53L0X_1_ADDR  0x21
-#define VL53L0X_2_ADDR  0x22
-#define VL53L0X_3_ADDR  0x23
-#define VL53L0X_4_ADDR  0x24
-
-VL53L0X_Dev_t SensorOne;
-VL53L0X_Dev_t SensorTwo;
-VL53L0X_Dev_t SensorThree;
-VL53L0X_Dev_t SensorFour;
-
-VL53L0X_Dev_t *pSensorOne = &SensorOne;
-VL53L0X_Dev_t *pSensorTwo = &SensorTwo;
-VL53L0X_Dev_t *pSensorThree = &SensorThree;
-VL53L0X_Dev_t *pSensorFour = &SensorFour;
-
-VL53L0X_RangingMeasurementData_t    SensorOneRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorTwoRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorThreeRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorFourRangingMeasurementData;
+#define VL53L0X_DEFAULT_ADDR 0x29
+#define NUM_SENSORS 4
+
+int VL53L0X_XSHUT_MCP23xx_IO[NUM_SENSORS];
+int VL53L0X_ADDR[NUM_SENSORS];
+VL53L0X_Dev_t Sensors[NUM_SENSORS];
+VL53L0X_Dev_t *pSensors[NUM_SENSORS];
+VL53L0X_RangingMeasurementData_t SensorsRangingMeasurementData[NUM_SENSORS];
+vl53l0x_driver::vl53l0x sensor_msg_array[NUM_SENSORS];
+int i2c_bus_instance;
+std::string i2c_bus_path;
+
+void initialize() {
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    VL53L0X_XSHUT_MCP23xx_IO[i] = i;
+    pSensors[i] = &Sensors[i];
+  }
+  for (int i = 0, j = 0x21; i < NUM_SENSORS; j++, i++)

```

Comment:
Can we do something like this:
```suggestion
for (int i = 0; i < NUM_SENSORS; i++) {
    VL53L0X_XSHUT_MCP23xx_IO[i] = i;
    pSensors[i] = &Sensors[i];
    VL53L0X_ADDR[i] = (0x21 + i);
}

Reply:
yes